File: client/java/src/main/java/com/be/mesh/client/mpc/MeshFlag.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.mpc;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Arrays;
import java.util.List;

/**
 * @author coyzeng@gmail.com
 */
@Getter
@AllArgsConstructor
public enum MeshFlag {

    HTTP("00", Consumer.HTTP),
    GRPC("01", Consumer.GRPC),
    MQTT("02", Consumer.MQTT),
    TCP("03", Consumer.TCP),

    JSON("00", Codec.JSON),
    PROTOBUF("01", Codec.PROTOBUF),
    XML("02", Codec.XML),
    THRIFT("03", Codec.THRIFT),
    YAML("04", Codec.YAML),

    ;

    public static final List<MeshFlag> PROTO = Arrays.asList(HTTP, GRPC, MQTT, TCP);
    public static final List<MeshFlag> CODEC = Arrays.asList(JSON, PROTOBUF, XML, THRIFT, YAML);
    private final String code;
    private final String name;

    public static MeshFlag ofProto(String code) {
        return PROTO.stream().filter(x -> x.code.equals(code)).findFirst().orElse(MeshFlag.HTTP);
    }

    public static MeshFlag ofCodec(String code) {
        return CODEC.stream().filter(x -> x.code.equals(code)).findFirst().orElse(MeshFlag.JSON);
    }

    public static MeshFlag ofName(String name) {
        return Arrays.stream(MeshFlag.values()).filter(x -> x.name.equals(name)).findFirst().orElse(MeshFlag.HTTP);
    }
}

File: client/python/mesh/kinds/meshflag.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#

from enum import Enum


class MeshFlag(Enum):
    HTTP = "00", "http", 0
    GRPC = "01", "grpc", 0
    MQTT = "02", "mqtt", 0
    TCP = "03", "tcp", 0

    JSON = "00", "json", 1
    PROTOBUF = "01", "protobuf", 1
    XML = "02", "xml", 1
    THRIFT = "03", "thrift", 1
    YAML = "04", "yaml", 1

    def get_code(self):
        return self.value[0]

    def get_name(self):
        return self.value[1]

    @staticmethod
    def of_proto(code: str) -> "MeshFlag":
        for member in MeshFlag:
            if member.value and member.value[2] == 0 and member.value[0] == code:
                return member
        return MeshFlag.HTTP

    @staticmethod
    def of_code(code: str) -> "MeshFlag":
        for member in MeshFlag:
            if member.value and member.value[2] == 1 and member.value[0] == code:
                return member
        return MeshFlag.JSON

    @staticmethod
    def of_name(name: str) -> "MeshFlag":
        for member in MeshFlag:
            if member.value and member.value[1] == name:
                return member
        return MeshFlag.HTTP
